src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,159 @@
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node currentNode = findIndex(index);
+        Node newNode = new Node(currentNode.previous, value, currentNode);
+        if (index == 0) {
+            head = newNode;
+        } else {
+            currentNode.previous.next = newNode;
+        }
+        currentNode.previous = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (!checkIndex(index)) {
+            throw new IndexOutOfBoundsException("Failed get from LinkedList. Please check index");
+        }
+        return findIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (!checkIndex(index)) {
+            throw new IndexOutOfBoundsException("Failed set in LinkedList. Please check index");
+        }
+        Node node = findIndex(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (!checkIndex(index)) {
+            throw new IndexOutOfBoundsException("Failed delete from LinkedList."
+                    + " Please check index");
+        }
+        Node node = findIndex(index);
+        if (node == tail && node == head) {
+            head = null;
+            tail = null;
+        } else if (node == tail) {
+            node.previous.next = null;
+            tail = node.previous;
+        } else if (node == head) {
+            node.next.previous = null;
+            head = node.next;
+        } else {
+            node.previous.next = node.next;
+            node.next.previous = node.previous;
+        }
+        size--;
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        if (isEmpty()) {
+            throw new IndexOutOfBoundsException("Cannot delete from empty list");
+        }
+        int index = getIndex(object);
+        if (index == -1) {
+            return false;
+        }
+        remove(index);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private class Node {
+        private Node previous;
+        private T value;
+        private Node next;
+
+        public Node(Node previous, T value, Node next) {
+            this.previous = previous;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node findIndex(int index) {
```

Comment:
let's iterate our LL from head to middle of tail to middle depending on the index